engine/src/main/java/org/camunda/bpm/engine/impl/batch/AbstractBatchJobHandler.java

```diff
@@ -169,6 +171,13 @@ protected void createJobEntities(BatchEntity batch, T configuration, String depl
       ByteArrayEntity configurationEntity = saveConfiguration(byteArrayManager, jobConfiguration);
 
       JobEntity job = createBatchJob(batch, configurationEntity);
+
+      if (jobConfiguration.getIds() != null && jobConfiguration.getIds().size() == 1) {
+        job.setProcessInstanceId(jobConfiguration.getIds().get(0));
+        String processDefinitionId = JsonUtil.getString(JsonUtil.asObject(batch.getConfigurationBytes()), PROCESS_DEFINITION_ID, null);
+        job.setProcessDefinitionId(processDefinitionId);
```

Comment:
❌ Considering the last discussions, let's populate the `processDefinitionId` field in `RestartProcessInstancesJobHandler#postProcessJob`
```suggestion
```